Hydra: updates to config composition info. (#4041)
iterative/dvc.org: content/docs/user-guide/experiment-management/hydra-composition.md

_New in DVC 2.25.0 (see `dvc version`)_

[Hydra](https://hydra.cc/) is a framework to configure complex applications. DVC
supports Hydra's [config composition] as a way to configure [experiment runs].

<admon type="info">

At the moment you must explicitly enable this feature with:

```dvc
$ dvc config hydra.enabled True
```

</admon>

[config composition]:
  https://hydra.cc/docs/tutorials/basic/your_first_app/composition/
[experiment runs]: /doc/user-guide/experiment-management/running-experiments

## How it works

> Upon `dvc exp run`, Hydra will be used to compose a single `params.yaml` before
> executing the experiment. Hydra config and existing <abbr>parameters</abbr> will
> be combined to configure the [underlying DVC pipeline](#running-experiments)
> that contains your experiment.

Reviewer: ```suggestion
Upon `dvc exp run`, Hydra will be used to compose a single `params.yaml` before
executing the experiment. The composed `params.yaml` will
be used to configure the [underlying DVC pipeline](#running-experiments)
that contains your experiment.
```

Minor: `Hydra config and existing parameters will be combined` seems a bit vague and confusing. How are they combined?

Author: Hmmm yeah I guess I though existing manual params.yaml files would also be taken into consideration, but this will completely overwrite those right? E.g. if you have a manual params.yaml file but later enable Hydra and setup its config groups and a default list.

Going with this for now:
```suggestion
Upon `dvc exp run`, Hydra will be used to compose a single `params.yaml` before
executing the experiment. Its <abbr>parameters</abbr> will configure the
[underlying DVC pipeline](#running-experiments) that contains your experiment.
```

Reviewer: > this will completely overwrite those right?

Just to confirm, that's correct.

    momentum: 0.9
```

## Running experiments

> <details>

Author: Still unsure of this title/ section. We have an entire page with the same name at https://dvc.org/doc/user-guide/experiment-management/running-experiments.

Author: Maybe we should try to combine or reorganize these topics (in the future). Rel. https://github.com/iterative/dvc.org/issues/2768 -- feel free to add a check box there.

Reviewer: Good idea! Added.
